packages/manager-react-components/src/components/datagrid/visibility/visibility-management.component.tsx

```diff
@@ -0,0 +1,89 @@
+import {
+  ODS_BUTTON_SIZE,
+  ODS_BUTTON_VARIANT,
+  ODS_ICON_NAME,
+  ODS_TEXT_PRESET,
+} from '@ovhcloud/ods-components';
+import {
+  OdsButton,
+  OdsCheckbox,
+  OdsFormField,
+  OdsPopover,
+  OdsText,
+} from '@ovhcloud/ods-components/react';
+import { useRef } from 'react';
+import { useTranslation } from 'react-i18next';
+
+export type ColumnsVisibility = {
+  id: string;
+  isDisabled: boolean;
+  label: string;
+  enableHiding: boolean;
+  isVisible: () => boolean;
+  onChange: () => void;
+};
+
+export type ColumnsVisibilityProps = {
+  columnsVisibility: ColumnsVisibility[];
+};
+
+export function VisibilityManagement({
+  columnsVisibility,
+}: Readonly<ColumnsVisibilityProps>) {
+  const { t } = useTranslation('datagrid');
+  const visibilityPopoverRef = useRef(null);
+  const numberVisibleColumns = columnsVisibility?.filter((col) =>
+    col.isVisible(),
+  ).length;
+  const disabledVisibleColumns = columnsVisibility?.filter(
+    (col) => col.enableHiding === false,
+  ).length;
+
+  return (
+    <div className="ml-[10px]">
```

Comment:
Since this styling can depend on Topbar use-case like add the margin if "Filters" button is displayed. So better to move this to Topbar component.

Reply:
it's fixed